src/Activity/TrainWithExercises/index.js

```diff
@@ -36,18 +44,24 @@ class TrainWithExercise extends React.Component {
         <ChooseExercise
           onSelectExercise={this.handleSelectExercise}
           onNavigationBackToMenu={this.props.getBackToMenu}
+          activityChosen = {this.props.activityChosen}
         />
       );
     }
+    let data = parallelogramData[this.state.index];
+    if (this.props.activityChosen === "mammals"){
+      data = mammalsData[this.state.index];
+    }
```

Comment:
It's easier to understand the code if you use `const` when you can avoid using `let`. If you define a variable with `let`, I am not sure what the value will be and I need to read the code carefully to understand it. If you define it with `const` then it's clearer.

You can use the trick that I mentionned to you earlier

```
const data = {
    mammals: mammalsData[this.state.index],
    parallelogram: parallelogramData[this.state.index] 
}[this.props.activityChosen]
```

Reply:
i tried to use this shorthand and didn't work for me. i found that data is always undefined. what could be the problem ?

Reply:
Look up what are the values of `activityChosen` and `index`. 
I cannot help you if you don't push your code